javalib/src/main/scala/java/io/FileDescriptor.scala

```diff
@@ -3,17 +3,26 @@ package java.io
 import scala.scalanative.posix.unistd
 
 /** Wraps a UNIX file descriptor */
-final class FileDescriptor private[io] (private[io] val fd: Int) {
+final class FileDescriptor private[io] (private[io] val fd: Int,
+                                        val readOnly: Boolean = false) {
 
   def this() = this(-1)
 
+  // inspired by Apache Harmony including filedesc.c
   def sync(): Unit =
-    unistd.fsync(fd) match {
-      case 0 => ()
-      case _ => throw new java.io.SyncFailedException("sync failed")
+    if (fd <= 2) throwSyncFailed()
```

Comment:
The POSIX spec says that `stdin`, `stdout` and `stderr` have value `0`, `1` and `2`, so that's ok 👍